v2/component/launcher.go

```diff
@@ -761,19 +791,32 @@ func (l *Launcher) openBucket() (*blob.Bucket, error) {
 	return blob.OpenBucket(context.Background(), l.bucketConfig.bucketURL())
 }
 
-func getMinioCredential() (*v1.Secret, error) {
-	restConfig, err := rest.InClusterConfig()
-	if err != nil {
-		return nil, fmt.Errorf("Failed to initialize kubernetes client: %w", err)
-	}
-	clientSet, err := kubernetes.NewForConfig(restConfig)
+func getMinioDefaultEndpoint() string {
+	minioHost := os.Getenv("MINIO_SERVICE_SERVICE_HOST")
+	minioPort := os.Getenv("MINIO_SERVICE_SERVICE_PORT")
+	if minioHost != "" && minioPort != "" {
+		// If there is a minio-service Kubernetes service in the same namespace,
+		// MINIO_SERVICE_SERVICE_HOST and MINIO_SERVICE_SERVICE_PORT env vars should
+		// exist by default, so we use it as default.
+		return minioHost + ":" + minioPort
+	}
+	// If the env vars do not exist, we guess that we are running in KFP multi user mode, so default minio service should be `minio-service.kubeflow:9000`.
+	glog.Info("Cannot detect minio-service in the same namespace, default to minio-service.kubeflow:9000 as MinIO endpoint.")
+	return "minio-service.kubeflow:9000"
+}
+
+func getLauncherConfig(clientSet *kubernetes.Clientset, namespace string) (*v1.ConfigMap, error) {
+	config, err := clientSet.CoreV1().ConfigMaps(namespace).Get(context.Background(), "kfp-launcher", metav1.GetOptions{})
 	if err != nil {
-		return nil, fmt.Errorf("Failed to initialize kubernetes client set: %w", err)
-	}
-	namespace := os.Getenv("KFP_NAMESPACE")
-	if namespace == "" {
-		return nil, fmt.Errorf("Env variable 'KFP_NAMESPACE' is empty")
+		if k8errors.IsNotFound(err) {
+			// LauncherConfig is optional, so ignore not found error.
```

Comment:
Can we add a log here indicate we can't find the configmap.

Reply:
Makes sense, added info log